Model/DutyCalculator.php

```diff
@@ -116,84 +116,333 @@ public function __construct(
+           //if this amount is more than 0 even though the user did not choose to apply it on
+           //computation of total landed cost/billing
+           $this->_logger->debug('In handleTaxApplicability method --- apply is false');
+        }
+
+        $quote->save();//this is needed so that different aspects related to a quote are available on other pages.
+        //More specifically saving quote in database is needed as apparently Magento session scopes are not application
+        // wide but code area specific (?)
+    }
+
+
+    /** handles when duty is not applicable (based on some configuration) for the country
+     * or shipping is not allowed for the country
+     * and saves proper state of quote so that the information is available beyond this page/area
+     * @param float $duty
+     * @param \Magento\Quote\Model\Quote $quote
+     */
+    public function handleNoDutyOrShippingCase($duty, $quote)
+    {
+        $this->response->setSuccess(true);
+        $this->response->setDuty($duty);
+        $this->checkoutSession->setReachDuty($duty);
+        $quote->setReachDuty($duty) ;
+        $this->_logger->debug('In duty or shipping not allowed section');
+        $quote->save();
+    }
+
+    /** Sets appropriate response (eventually be https response) values
+     * when state/province is (needed for proper duty calculation
+     * but) not yet specified
+     * @param float $duty
+     */
+    public function handleStateUnspecifiedCase($duty)
+    {
+       $quote = $this->checkoutSession->getQuote();
+       $this->response->setSuccess(true);
+       $this->response->setDuty($duty);
+       $this->_logger->debug('Special country where state selection is neccessary before initiating DHL API call;
+                but state is not selected anyway.');
+    }
+
+
+    /** Sets appropriate response (\Reach\Payment\Api\Data\DutyResponseInterface) values
+     *  when both state/province and country are specified
+     * @param Magento\Quote\Api\Data\AddressInterface $address
+     * @param bool $apply
+     */
+    public function handleCaseWithCountryAndStateSpecified($address, $apply)
+    {
+        $quote = $this->checkoutSession->getQuote();
+        $this->response->setDuty($quote->getReachDuty());
+        $this->handleTaxApplicability($address, $apply);
+        $this->_logger->debug('Country is specified but state is not selected (when both are needed); so we would'.
+            'prevent call to DHL API');
+    }
+
+
+    /** Uses already (recently) retrieved duty from DHL
+     * @param bool $apply
+     * @param \Magento\Quote\Model\Quote $quote
+     * @param Magento\Quote\Api\Data\AddressInterface $address
+     */
+    public function handleCaseWhereDutyIsAlreadyRetrieved($apply, $quote, $address)
+    {
+        if ($apply) {//user indicated that (s)he wants to apply duty and tax during calculation of billing
```

Comment:
ohhh I didn't know what this variable meant until this comment, maybe we can have a better description of what the variable represents in every line that it shows up in the function comments. For example, line 210, 197, 123. Or a more descriptive variable name.

Brainstorming: userPayFLC, payDutiesAndTaxesChecked, applyFLC...

Reply:
Shall be more than happy to discuss tomorrow @lizzynibali 
I did not know the meaning either - figured that out while playing with the code/tracing the code to fix MAG-70. We inherited this guy - it was there already. :D  
Thanks.

Reply:
ok that makes sense, we can discuss this separately from this ticket :) thanks!